### PR TITLE
refactored code and added the rest of the seasons

### PR DESCRIPTION
**Pull request by jimenasalinas**
Hello, this PR is an improvement to #8. There are 2 major changes:

1. TriMet sent the data for the 2 missing seasons: Fall and Winter, these seasons are now in the code and in our final data set.
2. I refactored the structure of my code to make use of more functions and make it easier to make changes in the future.

The same tests apply, the tests passed:

Screen Shot 2024-04-30 at 11 10 39 PM

### CODE REVIEW

File: ingestion/portland_ingest.py [hunk 1]
```diff
@@ -21,131 +21,164 @@
...
-] = "Summer"
 
-# combine Tri Met data with calendar dates in order to convert the data set
-# from day_type (sat, sun, or weekday) + season level into date level
+def preprocess_location_id(location_id):
+    """
+    Add a city prefix to make the location id more specific
+    for our data model
+    """
+    return f"Portland-{location_id}"
```
Comment (meganhmoore) [inline ingestion/portland_ingest.py:158]: I think for now we will handle city as one column and location id as another from the db perspective so no need to combine those for ingestion, but fine to leave for now until db stage.

**Review by meganhmoore — APPROVED** [review]
looks much cleaner and readable, thank you! I think there may be a few more changes to fit things into the expected structure of the db once we get to that point, but for now this seems like a good point to stop at.

File: ingestion/portland_ingest.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 """
```
Comment (JPMartinezClaeys) [inline ingestion/portland_ingest.py:1]: As a general comment for the file, nothing that will block merging but if we could add type hints for the functions that would be great to increase readability.

File: ingestion/portland_ingest.py [hunk 3]
```diff
@@ -21,131 +21,164 @@
...
-calendar_df["day_type"] = calendar_df["date"].dt.day_name()
-
-day_type_mapping = {"Saturday": "Saturday", "Sunday": "Sunday"}
-
-calendar_df["day_type"] = (
-    calendar_df["day_type"].map(day_type_mapping).fillna("Weekday")
-)
+def load_ridership_datasets(file_names):
+    """
+    Takes a dictionary in the form key: season_day type
```
Comment (JPMartinezClaeys) [inline ingestion/portland_ingest.py:73]: This is an example where I think type hints could be quite helpful as "file_names" is ambiguous and its easier to check with type hints what is expected.

File: ingestion/portland_ingest.py [hunk 4]
```diff
@@ -21,131 +21,164 @@
...
+        calendar_df["day_type"]
+        .map({"Saturday": "Saturday", "Sunday": "Sunday"})
+        .fillna("Weekday")
+    )
+    calendar_df["season"] = "Spring"
+
+    # label seasons based on date ranges
+    for season, (season_start, season_end) in season_ranges.items():
+        calendar_df.loc[
+            (calendar_df["date"] >= season_start) & (calendar_df["date"] < season_end),
```
Comment (JPMartinezClaeys) [inline ingestion/portland_ingest.py:135]: I would double check that season_start and season_end are correct.
If you create a datetime.datetime object for 2023.02.21 for example it creates an object representing 2023-02-21 00:00:00:00 so its important to check that it matches and I'm not sure it does with the current seasons dictionary.

**Review by JPMartinezClaeys — APPROVED** [review]
Looks good, as I mentioned in one of the comments I'm just slightly curious for the dates, but if that works feel free to merge

**Pull request merged**